Source/CarthageKit/Xcode.swift

```diff
@@ -1269,7 +1269,13 @@ private func stripBinary(binaryURL: NSURL, keepingArchitectures: [String]) -> Si
 public func copyProduct(from: NSURL, _ to: NSURL) -> SignalProducer<NSURL, CarthageError> {
 	return SignalProducer<NSURL, CarthageError>.attempt {
 		let manager = NSFileManager.defaultManager()
-
+		
+		if let path = to.path where
```

Comment:
Can you add a comment here describing why the check is needed and a link to this PR? It's not obvious why this would be necessary.